cfn-resources/autogen/schema-gen/main.go

```diff
@@ -0,0 +1,587 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/tidwall/pretty"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode"
+)
+
+//https://github.com/aws-cloudformation/cloudformation-cli/blob/master/src/rpdk/core/data/schema/provider.definition.schema.v1.json
+
+const (
+	url                = "https://github.com/aws-cloudformation/aws-cloudformation-rpdk.git"
+	MongoDBAtlasPrefix = "MongoDB::Atlas::"
+	Unique             = "Unique"
+)
+
+var optionalInputParams = []string{"envelope", "pretty", "apikeys", "app"}
+var optionalReqParams = []string{"app"}
+
+func main() {
+	//dir := "/autogen/schema-gen"
+	dir := "/schema-gen"
+
+	path, err := os.Getwd()
+	if err != nil {
+		fmt.Println(err)
+	}
+	fmt.Println(path)
+	dir = path + dir
+	file, err := ioutil.ReadFile(fmt.Sprintf("%s/mapping.json", dir))
+	if err != nil {
+		fmt.Printf("Path %s", path)
+		fmt.Printf("%v", err)
+		os.Exit(1)
+	}
+
+	doc, err := openapi3.NewLoader().LoadFromFile(fmt.Sprintf("%s/swagger.json", dir))
+	if err != nil {
+		fmt.Printf("Path %s", path)
+		fmt.Printf("%v", err)
+		os.Exit(1)
+	}
+
+	if doc == nil {
+		os.Exit(1)
+	}
+	// validate the swagger.yaml matches Openapi spec
+	_ = doc.Validate(context.Background())
+
+	data := OpenApiMapping{}
+	_ = json.Unmarshal(file, &data)
+
+	var h Handlers
+	json.Unmarshal([]byte(handler), &h)
+
+	chn := make(chan CfnSchema, len(doc.Components.Schemas))
+	reqFieldsChan := make(chan RequiredParams)
+
+	done := make(chan bool)
+	reqDone := make(chan bool)
+
+	go generateSchemas(chn, done)
+	go generateReqFields(reqFieldsChan, reqDone)
+	definitions := make(map[string]Definitions, 0)
+	var ids, readOnly, idsDef, readOnlyDef []string
+	var cfn CfnSchema
+	key := "params"
+	var typeName string
+	var description string
+	requiredParams := RequiredParams{}
+	createReqParams := []string{}
+	for _, res := range data.Resources {
+
+		queryParams := make(map[string]Property, 0)
+		allMethodProps := make(map[string]map[string]Property, 0)
+		bodySchema := make(map[string]map[string]Property, 0)
+
+		typeName = res.TypeName
+
```

Comment:
this is to find the tag and add the description
```suggestion
		t := doc.Tags.Get(ToSpaceCase(typeName))

		if t != nil {
			description = t.Description
		}
```

youll need to add the ToSpaceCase function

``` go
var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")

func ToSpaceCase(str string) string {
	snake := matchFirstCap.ReplaceAllString(str, "${1} ${2}")
	snake = matchAllCap.ReplaceAllString(snake, "${1} ${2}")
	return snake
}
```